### PR TITLE
testing/libtss2: new aport

### PR DESCRIPTION
**Pull request by whooo**

### CODE REVIEW

File: testing/libtss2/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+arch="all"
+makedepends="openssl-dev doxygen perl linux-headers cmocka-dev"
+depends=""
+subpackages="$pkgname-doc $pkgname-mu $pkgname-sys $pkgname-esys $pkgname-tcti-device $pkgname-tcti-mssim $pkgname-rc $pkgname-tctildr $pkgname-mu-static:mu_static $pkgname-sys-static:sys_static $pkgname-esys-static:esys_static $pkgname-tcti-device-static:device_static $pkgname-tcti-mssim-static:mssim_static $pkgname-rc-static:rc_static $pkgname-tctildr-static:tctildr_static $pkgname-dev"
+source="https://github.com/tpm2-software/tpm2-tss/releases/download/$pkgver/tpm2-tss-${pkgver}.tar.gz"
+builddir="$srcdir/tpm2-tss-${pkgver}"
+
+build() {
+    cd "$builddir"
+    ./configure CFLAGS="-fno-inline" \
```
[the remark this answers is not in the document]
Comment (Cogitri) [reply testing/libtss2/APKBUILD]: This still applies

File: testing/libtss2/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    pkgdesc="TPM 2.0 TCTI loader libraries"
+    mkdir -p "$subpkgdir/usr/lib"
+    mv "$pkgdir"/usr/lib/libtss2-tctildr.so* "$subpkgdir/usr/lib"
+}
+
+static() {
+    pkgdesc="TPM 2.0 (static libraries)"
+    mkdir -p "$subpkgdir/usr/lib"
+    mv "$pkgdir"/usr/lib/*.a "$subpkgdir/usr/lib"
+}
```
Comment (Cogitri) [inline testing/libtss2/APKBUILD]: abuild should do this automatically for you if you add $pkgname-static to subpackages

**Comment by Cogitri** [comment]
Please only indent with tabs. I'd recommend you to install atools for apkbuild-lint which can automatically find such errors.

Also: `testing/libtss2/APKBUILD:15:unnecesary usage of braces: ${pkgver}`

**Comment by asac** [comment]
Hi, if we want the more fine grained split that this package proposes over what we have in tpm2-tss package, could we maybe make this PR a PR against tpm2-tss instead of a new aport?

**Comment by Cogitri** [comment]
Oh, possibly, didn't realize it's the same package but more split up, sorry about that.

**Comment by whooo** [comment]
Created a new PR, https://github.com/alpinelinux/aports/pull/11909

**Comment by asac** [comment]
@whooo why are you making a new PR that moves? instead of making a PR that patches the existing tpm2-tss package and abandon this one... /me confused.

**Comment by whooo** [comment]
@asac I moved tpm2-tss and patched it afterwards in the new PR.
The reason I moved it was that (as I understand it) packages in community are not expected to be maintained for longer then six months and they aren't moved into main (which long term I'm looking for)

**Comment by asac** [comment]
lets first get a clean patch against the tpm2-tss package in without moving. @Cogitri moved the package to community for a reason as it was needed there, so lets not jump the gun and do too many things in one step. Also this patch still renames the package etc. can you please keep tpm2-tss as the main package name?

**Comment by whooo** [comment]
I don't really understand why it's needed it community, can't packages in community depend on packages in testing?

**Comment by Cogitri** [comment]
No

File: community/tpm2-tss/APKBUILD [hunk 3]
```diff
@@ -1,27 +1,86 @@
...
-license="BSD-2-Clause"
-makedepends="openssl-dev doxygen linux-headers cmocka-dev"
-subpackages="$pkgname-dev $pkgname-doc"
+makedepends="openssl-dev doxygen perl linux-headers cmocka-dev"
+subpackages="$pkgname-doc $pkgname-mu $pkgname-sys $pkgname-esys
+$pkgname-tcti-device $pkgname-tcti-mssim $pkgname-rc $pkgname-tctildr
+$pkgname-static $pkgname-dev"
 source="https://github.com/tpm2-software/tpm2-tss/releases/download/$pkgver/tpm2-tss-$pkgver.tar.gz"
-options="!check"
+builddir="$srcdir/tpm2-tss-$pkgver"
```
Comment (Cogitri) [inline community/tpm2-tss/APKBUILD]: No need to add this.

File: community/tpm2-tss/APKBUILD [hunk 4]
```diff
@@ -1,27 +1,86 @@
...
-pkgname="tpm2-tss"
-pkgver="2.3.1"
+# Contributor: Erik Larsson <who+alpine@cnackers.org>
+# Contributor: Alexander Sack <asac@pantacor.com>
+pkgname=tpm2-tss
+pkgver=2.3.1
 pkgrel=0
-pkgdesc="TPM2 tss library"
+pkgdesc="TPM 2.0 TSS"
+license="BSD-2-Clause"
```
Comment (Cogitri) [inline community/tpm2-tss/APKBUILD]: Just keep the license field at its old place

File: community/tpm2-tss/APKBUILD [hunk 5]
```diff
@@ -1,27 +1,86 @@
-# Contributor: Alexander Sack <asac@pantacor.com>
+# Maintainer: Erik Larsson <who+alpine@cnackers.org>
 # Maintainer: Alexander Sack <asac@pantacor.com>
-pkgname="tpm2-tss"
-pkgver="2.3.1"
+# Contributor: Erik Larsson <who+alpine@cnackers.org>
+# Contributor: Alexander Sack <asac@pantacor.com>
```
Comment (Cogitri) [inline community/tpm2-tss/APKBUILD]: Please just have one Maintainer and have it at the last line of this comment block (so the `Contributor` lines are above it).

**Pull request closed without merging**